File: src/ImportConfigurable.java
import com.intellij.openapi.options.SearchableConfigurable;
import com.intellij.openapi.project.Project;
import org.jetbrains.annotations.Nls;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class ImportConfigurable implements SearchableConfigurable {

	private final Project project;
	private final ImportConfig importConfig;

	public ImportConfigurable(@NotNull Project project) {
		this.project = project;
		this.importConfig = ImportConfig.getInstance(project);
	}

	@NotNull
	@Override
	public String getId() {
		return "preferences.RunConfConfigurable";
	}

	@Nls(capitalization = Nls.Capitalization.Title)
	@Override
	public String getDisplayName() {
		return "RunConf RunConfImporter";
	}

	@Nullable
	@Override
	public JComponent createComponent() {
		return ConfigUI.createPanel(project, importConfig);
	}

	@Override
	public boolean isModified() {
		return false;
	}

	@Override
	public void apply() {

	}

	@Override
	public void disposeUIResources() {
		if (!importConfig.hasDefault) {
			importConfig.setDefaultPath(null);
		}
	}
}

File: src/ImportConfigurable.kt
import com.intellij.openapi.options.SearchableConfigurable
import com.intellij.openapi.project.Project
import org.jetbrains.annotations.Nls
import javax.swing.JComponent

class ImportConfigurable(private val project: Project) : SearchableConfigurable {
    private val importConfig = ImportConfig.getInstance(project)

    override fun getId(): String {
        return "preferences.RunConfConfigurable"
    }

    @Nls(capitalization = Nls.Capitalization.Title)
    override fun getDisplayName(): String {
        return "RunConf RunConfImporter"
    }

    override fun createComponent(): JComponent? {
        return ConfigUI.createPanel(project, importConfig)
    }

    override fun isModified(): Boolean {
        return false
    }

    override fun apply() {

    }

    override fun disposeUIResources() {
        if (!importConfig.hasDefault) {
            importConfig.defaultPath = null
        }
    }
}
